src/evo/deterministicmns.cpp

```diff
@@ -1181,6 +1181,31 @@ void CDeterministicMNManager::CleanupCache(int nHeight)
 
 }
 
+[[nodiscard]] static bool EraseOldDBData(CDBWrapper& db, const std::vector<std::string>& db_keys)
+{
+    bool erased{false};
+    for(const auto& db_key : db_keys) {
+        CDBBatch batch(db);
+        auto it = std::unique_ptr<CDBIterator>(db.NewIterator());
+        auto firstKey = std::make_pair(db_key, uint256());
+        it->Seek(firstKey);
+        while (it->Valid()) {
+            decltype(firstKey) curKey;
+            if (!it->GetKey(curKey) || std::get<0>(curKey) != db_key) {
+                break;
+            }
+            batch.Erase(curKey);
+            erased = true;
+            it->Next();
+        }
+        if (erased) {
+            db.WriteBatch(batch);
+            LogPrintf("CDeterministicMNManager::%s -- done cleaning old data for %s\n", __func__, db_key);
```

Comment:
maybe better to have a log before `WriteBatch` for case if it would be failed - at least something would be in logs to point a place of failure.